### ISSUE TITLE
Polymorphic type annotations generalize weak polymorphic variables

### ISSUE THREAD

**Issue by thierry-martinez** [ocaml/ocaml#9856]
Polymorphic type annotations generalize weak polymorphic variables, making such programs type check and segfault:
```
let f x =
  let ref : type a . a option ref = ref None in
  ref := Some x;
  Option.get !ref

let () = print_string (f 0)
```
This program is rejected by OCaml 4.10 but is accepted by OCaml 4.11 and trunk. Bisection points to the commit c272447b1eff7829f42a8173059eb743754f1e4c.

**Comment by gasche** [comment]
Woops! Looks like an opportunity for a 4.11.1.

**Comment by lpw25** [comment]
Fixed by #9857
